core/math/projection.cpp

```diff
@@ -313,6 +314,59 @@ void Projection::set_perspective(real_t p_fovy_degrees, real_t p_aspect, real_t
 	*this = *this * cm;
 }
 
+void Projection::set_oblique(real_t p_fovy_degrees, real_t p_aspect, real_t p_z_near, real_t p_z_far, Quaternion p_oblique_plane, bool p_flip_fov) {
+	if (p_flip_fov) {
+		p_fovy_degrees = get_fovy(p_fovy_degrees, 1.0 / p_aspect);
+	}
+
+	real_t sine, cotangent, deltaZ;
+	real_t radians = Math::deg_to_rad(p_fovy_degrees / 2.0);
+
+	deltaZ = p_z_far - p_z_near;
+	sine = Math::sin(radians);
+
+	if ((deltaZ == 0) || (sine == 0) || (p_aspect == 0)) {
+		return;
+	}
+	cotangent = Math::cos(radians) / sine;
+
+	set_identity();
+
+	columns[0][0] = cotangent / p_aspect;
+	columns[1][1] = cotangent;
+	columns[2][2] = -(p_z_far + p_z_near) / deltaZ;
+	columns[2][3] = -1;
+	columns[3][2] = -2 * p_z_near * p_z_far / deltaZ;
+	columns[3][3] = 0;
+
+	// Here goes oblique magic!
+	// Eric Lengyel Solution: http://terathon.com/code/oblique.html
+	Quaternion q;
+
+	/* clang-format off */
+	q.x = ((
+		p_oblique_plane.x < 0 ? -1 :
+		p_oblique_plane.x > 0 ? +1 : 0
+		) + columns[2][0]
+	) / columns[0][0];
+
+	q.y = ((
+		p_oblique_plane.y < 0 ? -1 :
+		p_oblique_plane.y > 0 ? +1 : 0
+		) + columns[2][1]
+	) / columns[1][1];
```

Comment:
Updated code to:
```
q.x = (SIGN(p_oblique_plane.x) + columns[2][0]) / columns[0][0];
q.y = (SIGN(p_oblique_plane.y) + columns[2][1]) / columns[1][1];
```

from what I recall, when I originally wrote the first version of this camera almost 3 years ago the `SIGN` macro didn't exist. Either that or at the time I did not search thoroughly enough. Given how quickly I found it now however it could be that either it was the first case, or I'm just more familiar with the engine codebase that finding things is easier.